src/Sulu/Bundle/ResourceBundle/Controller/FilterController.php

```diff
@@ -69,15 +70,24 @@ function ($id) use ($locale) {
      */
     public function cgetAction(Request $request)
     {
-        if ($request->get('flat') == 'true') {
-            $list = $this->getListRepresentation($request);
+        // check if context exists and filters are enabled for the given context
+        $context = $request->get('context');
+        $features = $this->getManager()->getFeaturesForAlias($context);
+
+        if ($context && (!$features || array_search('filters', $features) === false)) {
+            $exc = new RestException('Context unknown or filters not enabled for context!');
```

Comment:
you should add two diffrent exception classes for this (`UnkownContextException` and `NotEnabledFeatureException`)